pkcs11/_pkcs11.pyx

```diff
@@ -235,6 +235,13 @@ class Token(types.Token):
 
         if user_pin is not None and so_pin is not None:
             raise ArgumentsBad("Set either `user_pin` or `so_pin`")
+        elif user_pin is not None and use_protected_auth:
+            raise ArgumentsBad("Set either `user_pin` or `use_protected_auth`")
+        elif so_pin is not None and use_protected_auth:
+            raise ArgumentsBad("Set either `so_pin` or `use_protected_auth`")
+        elif use_protected_auth:
+            pin = None
+            user_type = CKU_USER
```

Comment:
Can you use protected auth with the SO pin?

Reply:
Looking at the spec, it does appear that protected auth can be used with SO pins. If that's the case, we'll need a way to distinguish the type of user for protected auth sessions.

The api already distinguishes pins with `user_pin` and `so_pin` so we could distinguish protected_auth with something like `user_protected_auth` and `so_protected_auth`. 

e.g.
```
def open(self, rw=False, user_pin=None, so_pin=None, user_protected_auth=False, so_protected_auth=False):
```

Reply:
Maybe you should use a sentinel value like `PROTECTED_AUTH` and then pass `user_pin=PROTECTED_AUTH` to login.

Your code then becomes

```python
# types.py
PROTECTED_AUTH = object()
"""Indicate the pin should be supplied via an external mechanism (e.g. pin pad)"""

# _pkcs11.pyx
 ...
elif user_pin is PROTECTED_AUTH:
 ...
elif user_pin is not None:
 ...
```